neo/rawio/spikegadgetsrawio.py

```diff
@@ -55,10 +55,10 @@ def __init__(self, filename="", selected_streams=None):
 
         Notes
         -----
-        This file format has multiple version. New versions include the gain for scaling.
-        The current implementation does not contain this feature because we don't have
-        files to test this. So now the gain is "hardcoded" to 1, and so units are
-        not handled correctly.
+        This file format has multiple versions:
+            - Newer versions include the gain for scaling to microvolts [uV].
+            - If the scaling is not found in the header, it will be "hardcoded" to 1,
+              in which case the units are not handled correctly.
```

Comment:
I'm wondering while we are fixing this if we want to make this message a bit clear. For newer users of Neo it might not click that `get_analogsignal_chunk` will still work fine, but that the units would not be accurate for `rescale_signal_raw_to_float`? What do you think @alejoe91?

Reply:
Or again we could think about a warning since the end-user might not know whether or not their file has the scaling. For example how would they know whether or not their gain is correct?

Reply:
I agree that the user may not understand the implications based on this description;  I followed the language in the existing docstring.  I'm not so sure that this change will help much, though.  Actually your comment about `get_analogsignal_chunk` and `rescale_signal_raw_to_float` is more helpful.  How about adding that comment in a sentence after the bullet points?

Reply:
A warning that the data has not been scaled (and therefore has no units) is also a good idea!

Reply:
My other idea would be that we add an attribute like `self.is_scaleable` and then in the note we say check this attribute to know if the data has units or not. I'm not sure how SpikeInterface determines whether an extractor has units or not so we can wait for Alessio to weigh in to make sure this is smoothest for users just using neo but also people using spikeinterface.

Reply:
A warning would be great! And also an attribute like @zm711 suggested please

Reply:
Thanks for the feedback @zm711  and @alejoe91. 
So what I should add is:
- note in the docstring about functions that will (`get_analogsignal_chunk`) / will not (`rescale_signal_raw_to_float`) work
- warning is raised in the block if "spikeScalingToUv" is not found and set self.is_scaleable to False (otherwise set to True)

Is there a specific format for raising warnings in Neo?  Is the attribute self.is_scaleable uniform across these rawio readers?

Reply:
> Is there a specific format for raising warnings in Neo?

No. It's been a bit inconsistent. originally we were using a logger, but for me I'm fine with just `warnings.warn`. If @alejoe91 feels strongly for something else though he can definitely comment.

> Is the attribute self.is_scaleable uniform across these rawio readers?

No. It would just be for this rawio (in fact we would avoid it for others). The ideal solution would be to know the gain in all files, but if that is really not possible we need a workaround. So `is_scaleable` is a workaround that we will just patch in spikeinterface for the extractor for this io-only.

> note in the docstring about functions that will (get_analogsignal_chunk) / will not (rescale_signal_raw_to_float) work

To be precise it will work but it will not actually scale. So values will not become float voltages. (ie function should still work but shouldn't be used to get "real values".

Reply:
Great, @zm711. I implemented the above to the best of my understanding.